includes/class-linkbacks-mf2-handler.php

```diff
@@ -248,107 +397,67 @@ public static function get_entries( $mf_array ) {
+					if ( array_key_exists( 'author', $mf['properties'] ) ) {
+						return self::flatten_microformats( $mf['properties']['author'] );
 					}
 				}
+				if ( in_array( 'h-card', $mf['type'] ) ) {
+					return self::flatten_microformats( $mf );
+				}
 			}
 		}
+		if ( isset( $mf_array['rels']['author'] ) ) {
+			return $mf_array['rels']['author'];
+		}
 
 		return null;
 	}
 
-	/**
-	 * helper to find the correct h-entry node
-	 *
-	 * @param array $mf_array the parsed microformats array
-	 * @param string $target the target url
-	 *
-	 * @return array the h-entry node or false
-	 */
-	public static function get_representative_entry( $entries, $target ) {
-		// iterate array
-		foreach ( $entries as $entry ) {
-			// check properties
-			if ( isset( $entry['properties'] ) ) {
-				// check properties if target urls was mentioned
-				foreach ( $entry['properties'] as $key => $values ) {
-					// check "normal" links
-					if ( self::compare_urls( $target, $values ) ) {
-						return $entry;
-					}
 
-					// check included h-* formats and their links
-					foreach ( $values as $obj ) {
-						// check if reply is a "cite"
-						if ( isset( $obj['type'] ) && array_intersect( array( 'h-cite', 'h-entry' ), $obj['type'] ) ) {
-							// check url
-							if ( isset( $obj['properties'] ) && isset( $obj['properties']['url'] ) ) {
-								// check target
-								if ( self::compare_urls( $target, $obj['properties']['url'] ) ) {
-									return $entry;
-								}
-							}
-						}
-					}
-				}
-
-				// check properties if target urls was mentioned
-				foreach ( $entry['properties'] as $key => $values ) {
-					// check content for the link
-					if ( 'content' == $key &&
-						preg_match_all( '/<a[^>]+?' . preg_quote( $target, '/' ) . '[^>]*>([^>]+?)<\/a>/i', $values[0]['html'], $context ) ) {
-						return $entry;
-					} elseif ( 'summary' == $key &&
-						preg_match_all( '/<a[^>]+?' . preg_quote( $target, '/' ) . '[^>]*>([^>]+?)<\/a>/i',  $values[0], $context ) ) {
-						return $entry;
-					}
-				}
-			}
-		}
-
-		// return first h-entry
-		return $entries[0];
-	}
 
 	/**
 	 * check entry classes or document rels for post-type
 	 *
 	 * @param string $target the target url
-	 * @param array $entry the represantative entry
-	 * @param array $mf_array the document
+	 * @param array $propeties
+	 * @param array $rels
 	 *
 	 * @return string the post-type
 	 */
-	public static function get_entry_type( $target, $entry, $mf_array = array() ) {
+	public static function get_entry_type( $target, $properties, $rels ) {
 		$classes = self::get_class_mapper();
 
 		// check properties for target-url
-		foreach ( $entry['properties'] as $key => $values ) {
+		foreach ( $properties as $key => $values ) {
 			// check u-* params
 			if ( in_array( $key, array_keys( $classes ) ) ) {
 				// check "normal" links
-				if ( self::compare_urls( $target, $values ) ) {
+				//	if ( self::compare_urls( $target, $values ) ) {
```

Comment:
It has to be checked, because this example for target: `http://example.com/webmention/target/placeholder` is returning `reply` but it should return `like`.

```HTML
<span class="h-entry">
  <a class="u-in-reply-to" href="http://example.com/">A Cool Post</a><a class="u-url" href="/"></a>
  <a class="u-like-of" href="http://example.com/webmention/target/placeholder">A Cool Post</a><a class="u-url" href="/"></a>
  <a class="u-in-reply-to" href="http://example.com/">A Cool Post</a><a class="u-url" href="/"></a>
</span>
```